internal/docstore/mongodocstore/mongo.go

```diff
@@ -142,16 +142,33 @@ func (c *collection) replace(ctx context.Context, a *driver.Action, upsert bool)
 	if err != nil {
 		return err
 	}
-	opts := options.Replace()
-	if upsert {
-		opts.SetUpsert(true) // Document will be created if it doesn't exist.
+	rev, err := a.Doc.GetField(docstore.RevisionField)
+	if err != nil && gcerrors.Code(err) != gcerrors.NotFound {
```

Comment:
Do all other error codes need to proceed?

Reply:
I think if somehow we get an error that isn't NotFound, then something is really wrong with the document and we should fail.